src/history/hs_rec.c

```diff
@@ -559,20 +523,13 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
             /* We can only delete history store entries that have timestamps. */
             WT_ERR(__wt_hs_delete_key_from_ts(session, btree->id, key, 1));
             WT_STAT_CONN_INCR(session, cache_hs_key_truncate_mix_ts);
-            clear_hs = false;
```

Comment:
Sorry to mention it here. The following comment needs to change.

                /*
                 * Don't insert updates with timestamps after updates without timestamps to the
                 * history store.
                 */